python/triton/runtime/autotuner.py

```diff
@@ -102,14 +102,19 @@ def kernel_call():
             if config.pre_hook:
                 config.pre_hook(full_nargs)
             self.pre_hook(args)
-            self.fn.run(
-                *args,
-                num_warps=config.num_warps,
-                num_stages=config.num_stages,
-                num_ctas=config.num_ctas,
-                **current,
-            )
-            self.post_hook(args)
+            try:
+                self.fn.run(
+                    *args,
+                    num_warps=config.num_warps,
+                    num_stages=config.num_stages,
+                    num_ctas=config.num_ctas,
+                    **current,
+                )
+            except Exception as e:
+                self.post_hook(args, exception=e)
```

Comment:
I feel like you might want to handle exceptions that occur inside post_hook, so that any exception raised by `run` is guaranteed to be thrown?

```
except Exception as e:
  try:
    self.post_hook()
  finally:
    raise # I assume / hope this raises `e` and not the exception raised by post_hook
```